pkg/controllers/webhookauthenticator/webhookauthenticator_controller.go

```diff
@@ -176,3 +165,56 @@ func (c *webhookAuthenticatorController) ensureKubeConfigSecret(ctx context.Cont
 	secret, _, err := resourceapply.ApplySecret(c.secrets, recorder, requiredSecret)
 	return secret, err
 }
+
+func (c *webhookAuthenticatorController) getAuthenticatorCertKeyPair(ctx context.Context) (key, cert []byte, err error) {
+	var waitingForCertKey *string
+	defer func() {
+		if waitingForCertKey != nil {
+			cond := operatorv1.OperatorCondition{
+				Type:    "AuthenticatorCertKeyProgressing",
+				Status:  operatorv1.ConditionFalse,
+				Reason:  "AsExpected",
+				Message: "All is well",
+			}
+
+			if len(*waitingForCertKey) > 0 {
+				cond.Status = operatorv1.ConditionTrue
+				cond.Reason = "WaitingForCertKey"
+				cond.Message = *waitingForCertKey
+			}
+
+			if _, _, statusErr := v1helpers.UpdateStatus(c.operatorClient, v1helpers.UpdateConditionFn(cond)); statusErr != nil {
+				klog.Error("failed to update operator status: %v", statusErr)
```

Comment:
we don't want to return `statusErr` instead?

Reply:
good point